### PR TITLE
fix: format currency

### PR DESCRIPTION
**Pull request by lelenei**
## Description

Closes mf-2793

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/web3-shared/base/src/utils/currency.ts [hunk 1]
```diff
@@ -97,10 +97,7 @@ export function formatCurrency(
                 case 'currency':
                     return resolvedSymbols[value] ?? value
                 case 'fraction':
-                    const unFormatString = decimalValue.toFormat(isMoreThanOrEqualToOne ? 2 : 6).replace('0.', '')
-                    return isLessMinValue || bgValue.isGreaterThanOrEqualTo(1) || isMoreThanOrEqualToOne
-                        ? unFormatString
-                        : unFormatString.replace(/(0+)$/, '')
+                    return value
```
Comment (guanbinrui) [inline packages/web3-shared/base/src/utils/currency.ts]: CC @zhouhanseng @Lanttcat

Comment (hanszhoou) [reply packages/web3-shared/base/src/utils/currency.ts]: Unit test not passed.

Comment (Lanttcat) [reply packages/web3-shared/base/src/utils/currency.ts]: And more detail, please view mf-1150

File: packages/web3-shared/base/src/utils/currency.ts [hunk 2]
```diff
@@ -91,16 +91,31 @@ export function formatCurrency(
 
     if (symbol) return `${bgValue.toNumber()} ${symbol}`
 
+    if (bgValue.isLessThan(1)) {
+        return digitalCurrencyModifier(formatter.formatToParts(bgValue.toNumber()), resolvedSymbols)
```
Comment (hanszhoou) [inline packages/web3-shared/base/src/utils/currency.ts]: Keep it DRY, there're two references of `digitalCurrencyModifier`.

File: packages/web3-shared/base/src/utils/currency.ts [hunk 3]
```diff
@@ -91,16 +91,31 @@ export function formatCurrency(
 
     if (symbol) return `${bgValue.toNumber()} ${symbol}`
 
+    if (bgValue.isLessThan(1)) {
```
Comment (hanszhoou) [inline packages/web3-shared/base/src/utils/currency.ts]:
```suggestion
    if (!isMoreThanOrEqualToOne) {
```

File: packages/web3-shared/base/src/utils/currency.ts [hunk 4]
```diff
@@ -91,6 +91,24 @@ export function formatCurrency(
...
+                    case 'currency':
+                        return resolvedSymbols[value] ?? value
+                    default:
+                        return value
+                }
+            })
+            .join('')
+    }
+
     return digitalCurrencyModifier(formatter.formatToParts(bgValue.toNumber()), resolvedSymbols)
```
Comment (hanszhoou) [inline packages/web3-shared/base/src/utils/currency.ts]: DRY

**Review by hanszhoou — APPROVED** [review]

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**